### ISSUE TITLE
[proposal] Ignore match when the alternative is the same

### ISSUE THREAD

**Issue by Mogztter** [sapegin/textlint-rule-stop-words#16]
I'm using this rule as a replacement of [Vale's `substitution` rule](https://docs.errata.ai/vale/styles#substitution).

I'm using the following definition to make sure that "Asciidoctor" and "AsciiDoc" have the correct casing: 

```json
{
  "rules": {
    "stop-words": {
      "defaultWords": false,
      "skip": ["Blockquote"],
      "words": [
        ["asciidoctor", "Asciidoctor"],
        ["asciidoc", "AsciiDoc"]
      ]
    }
  }
}
```

Since this rule allows to define replacement/alternative. I think it would be useful to check that the replacement/alternative is not strictly equals to the match.

Under this proposal, the following sentence will report an error:

```
AsciiDoctor is great!
     ^
```

But the following will not:

```
Asciidoctor is great!
```

**Comment by sapegin** [comment]
I think I'm missing something here. Isn't `Asciidoctor` is what you've defined as the replacement? So the last example is already correct, and not reported because of that?

**Comment by Mogztter** [comment]
Sorry I wasn't clear, the last part is the expected result (not the actual/current result). Currently, both `Asciidoctor` and `AsciiDoctor` will be reported as error. With my proposal only `AsciiDoctor` will be reported as an error.
Does it make sense?

**Comment by sapegin** [comment]
Ah, yeah, that makes sense!
